### PR TITLE
Add support for night mode

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@samtstern Dang, I completely forgot about the [night mode issue](https://github.com/firebase/FirebaseUI-Android/issues/502) but the styling PR was the best desision ever! 😄 It just works™️

Here are a bunch of screenshots I took while testing, everything LGTM!
screenshot_20170612-123707
screenshot_20170612-123714
screenshot_20170612-123717
screenshot_20170612-123728
screenshot_20170612-123745
screenshot_20170612-123756
screenshot_20170612-123819
screenshot_20170612-123910
screenshot_20170612-123938
screenshot_20170612-123942
screenshot_20170612-124230
screenshot_20170612-124241
screenshot_20170612-123505

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
@samtstern I updated the base for this PR to 2.1

**Comment by SUPERCILEX** [comment]
@samtstern what are your thoughts on this?

File: app/src/main/java/com/firebase/uidemo/auth/LeakCatcher.java [hunk 1]
```diff
@@ -2,13 +2,18 @@
...
+import android.support.v7.app.AppCompatDelegate;
 
 import com.squareup.leakcanary.LeakCanary;
 import com.squareup.leakcanary.RefWatcher;
 
 public class LeakCatcher extends Application {
     private RefWatcher mRefWatcher;
 
+    static {
+        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_AUTO);
```
Comment (samtstern) [inline app/src/main/java/com/firebase/uidemo/auth/LeakCatcher.java]: Feels a little strange to do this in a class called 'LeakCatcher'.  Also should we use 'MODE_NIGHT_SYSTEM' to respect the system setting?

Comment (SUPERCILEX) [inline app/src/main/java/com/firebase/uidemo/auth/LeakCatcher.java]: @samtstern Same, I was just being lazy. 😄

**Comment by SUPERCILEX** [comment]
@samtstern As for the whole night mode thing, I've been really getting into AppCompat's coolness and I found these great resources which should clear things up for you:
- https://medium.com/@chrisbanes/appcompat-v23-2-daynight-d10f90c83e94
- Scroll down to DayNight https://android-developers.googleblog.com/2016/02/android-support-library-232.html

**Comment by SUPERCILEX** [comment]
## TL;DR

image
-
image
-
image
-
image
